hs_geographic_feature_resource/receivers.py

```diff
@@ -0,0 +1,522 @@
+                    uploaded_file_type=rslt['uploaded_file_type']
+                    uploadedFileCount=rslt['uploadedFileCount']
+                    uploadedFilenameString=rslt['uploadedFilenameString']
+                    shp_full_path=rslt['shp_full_path']
+                    if uploaded_file_type == "shp" or uploaded_file_type == "zipped_shp":
+                        meta_array, meta_dict=parse_shp_zshp(uploaded_file_type, baseFilename, uploadedFileCount, uploadedFilenameString, shp_full_path)
+                        metadata.extend(meta_array)
+            else:
+                validate_files_dict['are_files_valid'] = False
+                validate_files_dict['message'] = 'Please upload valid file(s).'
+        except:
+            validate_files_dict['are_files_valid'] = False
+            validate_files_dict['message'] = 'Uploaded files are invalid or corrupt.'
+        finally:
+            if tmp_dir is not None:
+                shutil.rmtree(tmp_dir)
+                tmp_dir = None
+
+
+# This handler is executed only when a metadata element is added as part of editing a resource
+@receiver(pre_metadata_element_create, sender=GeographicFeatureResource)
+def metadata_element_pre_create_handler(sender, **kwargs):
+    element_name = kwargs['element_name'].lower()
+    request = kwargs['request']
+
+    if element_name == 'originalcoverage':
+        element_form = OriginalCoverageValidationForm(data=request.POST)
+    elif element_name == 'fieldinformation':
+        element_form = FieldInformationValidationForm(data=request.POST)
+    elif element_name == 'geometryinformation':
+        element_form = GeometryInformationValidationForm(data=request.POST)
+
+    if element_form.is_valid():
+        return {'is_valid': True, 'element_data_dict': element_form.cleaned_data}
+    else:
+        return {'is_valid': False, 'element_data_dict': None}
+
+
+# This handler is executed only when a metadata element is added as part of editing a resource
+@receiver(pre_metadata_element_update, sender=GeographicFeatureResource)
+def metadata_element_pre_update_handler(sender, **kwargs):
+    element_name = kwargs['element_name'].lower()
+    request = kwargs['request']
+
+    if element_name == 'originalcoverage':
+        element_form = OriginalCoverageValidationForm(data=request.POST)
+    elif element_name == 'fieldinformation':
+        element_form = FieldInformationValidationForm(data=request.POST)
+    elif element_name == 'geometryinformation':
+        element_form = GeometryInformationValidationForm(data=request.POST)
+
+    if element_form.is_valid():
+        return {'is_valid': True, 'element_data_dict': element_form.cleaned_data}
+    else:
+        return {'is_valid': False, 'element_data_dict': None}
+
+
+@receiver(pre_delete_file_from_resource, sender=GeographicFeatureResource)
+def geofeature_pre_delete_file_from_resource(sender, **kwargs):
+    if sender is GeographicFeatureResource:
```

Comment:
if you are not attaching this receiver to any other sender, then you don't need to check it